components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -391,6 +417,24 @@ protected void buildClaims(AuthenticationContext context, Map<String, Object> js
         }
     }
 
+    /**
+     * Prefix give ClaimDialactUri to given claimUri.
+     */
+    private String getEffectiveClaimUri(String claimDialectUri, String claimUri) {
+
+        if (shouldPrefixClaimDialectUri() && StringUtils.isNotBlank(getClaimDialectURI())) {
+            return claimDialectUri + "/" + claimUri;
+        }
+        return claimUri;
+    }
+
+    public boolean shouldPrefixClaimDialectUri() {
+
+        Map<String, String> parameters = readParametersFromAuthenticatorConfig();
+        return parameters != null && parameters.containsKey(FacebookAuthenticatorConstants
+                .PREFIE_CLAIM_DIALECT_URI_PARAMETER);
```

Comment:
here .containsKey is not enough right? 
should use Boolean.paserBoolean(PREFIE_CLAIM_DIALECT_URI_PARAMETER-value) isn't it?

Reply:
done